docs(misconf): Add guide on input schema

simar7 opened the pull request:
## Description

Add documentation for defining custom input schemas.

Signed-off-by: Simar <simar@linux.com>

itaysk commented on docs/docs/misconfiguration/custom/schema.md:
```diff
@@ -0,0 +1,77 @@
+# Input Schema
+
+## Overview
+Policies can be defined with custom schemas that allow inputs to be verified against it. Adding an policy schema
```
```suggestion
Policies can be defined with custom schemas that allow inputs to be verified against them. Adding a policy schema
```

itaysk commented on docs/docs/misconfiguration/custom/schema.md:
```diff
@@ -0,0 +1,77 @@
+Policies can be defined with custom schemas that allow inputs to be verified against it. Adding an policy schema
+enables Trivy to show more detailed error messages when an invalid input is encountered.
+
+For instance if we have a policy that checks for misconfigurations in a `Dockerfile`, we could define the
```
Dockerfile is a great example for input schemas because dockerfiles are not json compatible by nature and without json schema it is impossible to know what you are even evaluating. I think it's worth mentioning and even reframing the intro to emphasize this.

simar7 replied:
added more context to show the benefit of input schemas

itaysk reviewed (commented):
great addition

knqyf263 reviewed (approved):
LGTM. I'll wait for Itay.

itaysk reviewed (approved):
LGTM, there's an existing mention to schemas [here](https://aquasecurity.github.io/trivy/v0.37/docs/misconfiguration/custom/#schemas). it might be better to remove it now and link to the full page

knqyf263 commented:
>it might be better to remove it now and link to the full page

I've updated it. https://github.com/aquasecurity/trivy/pull/3692/commits/8a7ffb3a88eb82e9fed91462a5a45777d1c8f93f

But I also found other places mentioning schemas. Don't we need to fix them?
https://github.com/aquasecurity/trivy/blob/8a7ffb3a88eb82e9fed91462a5a45777d1c8f93f/docs/docs/misconfiguration/custom/index.md?plain=1#L42-L43
https://github.com/aquasecurity/trivy/blob/8a7ffb3a88eb82e9fed91462a5a45777d1c8f93f/docs/docs/misconfiguration/custom/index.md?plain=1#L131

AnaisUrlichs commented on docs/docs/misconfiguration/custom/schema.md:
```diff
@@ -0,0 +1,93 @@
+    # METADATA
+    # schemas:
+    # - input: schema["dockerfile"]
+    package mypackage
```
why is it called `mypackage`?

simar7 replied:
Just a name.

simar7 commented:
> > it might be better to remove it now and link to the full page
> 
> I've updated it. [8a7ffb3](https://github.com/aquasecurity/trivy/commit/8a7ffb3a88eb82e9fed91462a5a45777d1c8f93f)
> 
> But I also found other places mentioning schemas. Don't we need to fix them?
> 
> https://github.com/aquasecurity/trivy/blob/8a7ffb3a88eb82e9fed91462a5a45777d1c8f93f/docs/docs/misconfiguration/custom/index.md?plain=1#L42-L43
> 
> 
> https://github.com/aquasecurity/trivy/blob/8a7ffb3a88eb82e9fed91462a5a45777d1c8f93f/docs/docs/misconfiguration/custom/index.md?plain=1#L131

I updated them.
